### PR TITLE
[SYCL][Bindless][Exp] Remove unneeded function argument causing memory leak in image create functions

### PR DESCRIPTION
**Pull request by DBDuncan**
Remove pi_mem argument from piextMemUnsampledImageCreate, piextMemSampledImageCreate and the corresponding UR functions. The argument is not used for any useful functionality and causes a memory leak.

### CODE REVIEW

**Comment by DBDuncan** [comment]
Corresponding UR PR: https://github.com/oneapi-src/unified-runtime/pull/1512

**Review by hdelan — APPROVED** [review]
LGTM

**Review by PietroGhg — APPROVED** [review]
Native CPU lgtm, thank you

**Review by isaacault — APPROVED** [review]

**Comment by MartinWehking** [comment]
LGTM

**Review by przemektmalon — APPROVED** [review]

**Review by Seanst98 — APPROVED** [review]

**Review by cppchedy — APPROVED** [review]

**Review by aelovikov-intel — APPROVED** [review]

**Comment by kbenzie** [comment]
https://github.com/intel/llvm/pull/13503 which fixes the ASAN test failures [here](https://github.com/intel/llvm/actions/runs/9064545616/job/24903608134?pr=13364) have now been merged, pull in the latest sycl branch changes to get the check passing.

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -94,24 +94,18 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
-  #     [DeviceSanitizer] Check out-of-bounds on sycl::local_accessor
-  set(UNIFIED_RUNTIME_TAG 4c69624c3f651819210148ad11d0e1e5c9af12de)
+  set(UNIFIED_RUNTIME_REPO "https://github.com/DBDuncan/unified-runtime.git")
+  # commit bbb04b65b59696697bcf04c7a6d843a9ea330b52
+  # Author: Duncan Brawley <duncan.brawley@codeplay.com>
+  # Date:   Thu, 21 Mar 2024 16:35:42 +0000
+  #    [Bindless][Exp] Remove phMem argument from bindless image creation functions
+  #     Remove phMem argument from bindless image creation functions.
+  #     It not used outside of UR and is currently causing a minor memory leak.
+  set(UNIFIED_RUNTIME_TAG bbb04b65b59696697bcf04c7a6d843a9ea330b52)
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:104]:
```suggestion
  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
  # commit 056d653264034e546d8b8f493e1d9f65c697829b
  # Merge: b7c89302 bbb04b65
  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
  # Date:   Fri May 17 15:11:12 2024 +0100
  #     Merge pull request #1512 from DBDuncan/duncan/fix_pi_mem_leak
  #     [Bindless][Exp] Remove phMem argument from bindless image creation functions
  set(UNIFIED_RUNTIME_TAG 056d653264034e546d8b8f493e1d9f65c697829b)
```

**Review by kbenzie — COMMENTED** [review]
https://github.com/oneapi-src/unified-runtime/pull/1512 has been merged please update the UR repo/tag as suggested and pull in the latest sycl branch changes.

**Review by kbenzie — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers please merge

**Comment by sarnex** [comment]
We aren't merging any PRs today because the CI is in a bad state, we will merge this on Monday.

**Comment by DBDuncan** [comment]
@sarnex Is CI in a better state now?

**Comment by aelovikov-intel** [comment]
> @sarnex Is CI in a better state now?

Let's see :)

**Pull request merged**